services/api/src/lib/utils/__tests__/schema.test.js

```diff
@@ -143,5 +189,76 @@ describe('createSchema', () => {
       expect(data.user.password).toBeUndefined();
     });
 
+    it('should not allow access to private autopoulated fields by default', async () => {
+      const User = createModel(createSchema({
+        secret: { type: String, access: 'private' },
+      }));
+
+      const shopSchema = createSchema({
+        user: {
+          ref: User.modelName,
+          type: mongoose.Schema.Types.ObjectId,
+          autopopulate: true,
+        }
+      });
+
+      shopSchema.plugin(require('mongoose-autopopulate'));
+      const Shop = createModel(shopSchema);
+
+      const user = new User({
+        secret: 'foo'
+      });
+      await user.save();
+
+      let shop = new Shop();
+      shop.user = user;
+      await shop.save();
+
+      shop = await Shop.findById(shop.id);
+
+      const data = shop.toObject();
+
+      expect(data.user.id).toBe(user.id);
+      expect(data.user._id).toBeUndefined();
+      expect(data.user.__v).toBeUndefined();
+      expect(data.user.secret).toBeUndefined();
+    });
+
+    it('should allow access to private autopoulated fields with options', async () => {
+      const User = createModel(createSchema({
+        secret: { type: String, access: 'private' },
+      }));
+
+      const shopSchema = createSchema({
+        user: {
+          ref: User.modelName,
+          type: mongoose.Schema.Types.ObjectId,
+          autopopulate: true,
+        }
+      });
+
+      shopSchema.plugin(require('mongoose-autopopulate'));
+      const Shop = createModel(shopSchema);
+
+      const user = new User({
+        secret: 'foo'
+      });
+      await user.save();
+
+      let shop = new Shop();
+      shop.user = user;
+      await shop.save();
+
+      shop = await Shop.findById(shop.id);
+
+      const data = shop.toObject({
```

Comment:
The only thought I have is if this could benefit from being more verbose.
As its pretty important field. 

shop.toObject(accessLevel, { ...options }) or if no accessed required shop.toObject({...options})
where accessLevel is a string.

same for toJSON.

Reply:
`toObject` is a mongoose method that doesn't have that signature so I'm not really sure what you mean... it does take options that are passed down to child documents..

I feel like having `{ private: true }` is quite explicit and default behavior is NOT to expose private fields without this which I like as well

Reply:
kk